### ISSUE TITLE
Style customization using head.html

### ISSUE THREAD

**Issue by srid** [EyebrowHairs/garden#6]
FYI. You can create a file named `head.html`, and neuron will inject that into the `<head>` tag of all generated pages. This way you can inject any custom JavaScript or CSS to your pages. For example, https://darkmodejs.learn.uno/

**Comment by EyebrowHairs** [comment]
Wooooah nice! Time to experiment now!

**Comment by srid** [comment]
eg: 

https://truong.io/notes/
https://github.com/leonardt/notes/blob/master/head.html

**Comment by EyebrowHairs** [comment]
Nice! Does it work to link to stylesheets too?

**Comment by srid** [comment]
Yup, you can put your stylesheet, say style.css, under the "static" folder,
and then link to it using the relative URL "./static/style.css". You can do
the same for JS and other files.

**Comment by srid** [comment]
Still in brainstorming stage, but it might just be possible to have neuron (or a new tool) to support full-themeing, allowing users to write their own HTML/CSS. I'm trying to figure out a template language for that, and this one seems like a good fit. I'm just curious what you think of `heist` @EyebrowHairs -- from user's point of view. It looks like you can write 'templates' and re-use them in multiple pages (see bind and apply). http://snapframework.com/docs/tutorials/heist

**Comment by EyebrowHairs** [comment]
Oh interesting! Is it a similar idea to something like "Includes" for Jekyll (https://jekyllrb.com/docs/includes/)? I wasn't sure what the difference is between <apply> and <apply-content> but the general concept seems straightforward.

**Comment by srid** [comment]
Think I'll go with https://mustache.github.io/mustache.5.html

You can see an example here: https://github.com/srid/emanote/blob/d73814569eb7fadc5bdb1945b767f6bc9cd63681/doc/index.html#L1

**Comment by EyebrowHairs** [comment]
Looks much simpler and cleaner!

**Comment by srid** [comment]
Another example of using head.html to customize CSS, JavaScript, etc

* Demo: https://alexsoto.dev/
* Source: https://github.com/alex-a-soto/zettelkasten/blob/master/head.html

(You can add background pattern too, https://stream.srid.ca/ )

**Comment by EyebrowHairs** [comment]
Ooh, fancy! I still have to work on a theme, but I guess my general grand idea is to create a 'regular' looking homepage, with the Neuron interface once you click on the 'notes/blog' portion of the site.

Just another thing...does it work to create a subfolder of notes, or should I use nested tags/folgezettel links?

**Comment by srid** [comment]
Subfolders should work.

There is also the related plugin: https://neuron.zettel.page/dirtree

**Comment by EyebrowHairs** [comment]
Hmm, so how does it work to create a separate homepage? Should it be in another repository?

**Comment by srid** [comment]
> Should it be in another repository?

That's the easiest way, yes. Just click "Use this template" from https://github.com/srid/neuron-template to create a new site.

(You can use gitsubmodules to link the new repository to your main zettelkasten repo, but this means you have to make sure that filenames are unique across the both repos; already "index.md" will conflict).

**Comment by EyebrowHairs** [comment]
I see! I was looking at the example you shared earlier:

>https://truong.io/notes/
https://github.com/leonardt/notes/blob/master/head.html

So looking at that, I thought the homepage and notes page were separate. 
Would it work at all to change the BaseURL to "___.com/notes"? I might just use some Jekyll theme or something for the homepage and copy over the header/footer so it looks consistent (if it works like that lol).

**Comment by srid** [comment]
neuron generates only relative URLs, so yes - you can just move the generated .html files to a subfolder and it should work. Just change the base URL in `neuron.dhall` to the correct URL so the metadata links in `<head>` remain functional.

**Comment by srid** [comment]
you probably want to add `-o .neuron/output/notes` to the command here: https://github.com/EyebrowHairs/garden/blob/12adcc40a2946575926dd2c7d6ddd40df97127af/.github/workflows/publish.yaml#L22

And then copy over your jekyll generated .html to `.neuron/output/`

**Comment by EyebrowHairs** [comment]
Or alternatively...maybe I'll try to somewhat recreate this theme with neuron: https://github.com/Trybnetic/minimalist

**Comment by srid** [comment]
@EyebrowHairs checkout https://tailwindcss.com/ - great CSS framework that is easy to use. you can use tailwind's classes directly in neuron markdown files.

**Comment by EyebrowHairs** [comment]
Oh interesting, I'll check it out! Is it similar to using Semantic UI in the MD file? As an example, how can I replicate this [message box](https://semantic-ui.com/collections/message.html)?

**Comment by srid** [comment]
@EyebrowHairs I just used Tailwind to create a nice looking pink message box in https://www.srid.ca/now

To see how I did it, check out the just-added neuron docs: https://neuron.zettel.page/tailwind

**Comment by srid** [comment]
https://tribe.srid.ca/@srid/posts/A62qdzCRqf3tdKsONU
